com.unity.cinemachine/Runtime/Debug/CinemachineDebug.cs

```diff
@@ -0,0 +1,149 @@
+using System;
+using UnityEngine;
+using System.Collections.Generic;
+using System.Text;
+#if UNITY_EDITOR && CINEMACHINE_UIELEMENTS
+using UnityEditor;
+using UnityEngine.UIElements;
+#endif
+
+namespace Unity.Cinemachine
+{
+    /// <summary>Manages onscreen positions for Cinemachine debugging output</summary>
+    static class CinemachineDebug
+    {
+        static List<StringBuilder> s_AvailableStringBuilders;
+
+#if CINEMACHINE_UNITY_IMGUI
+        /// <summary>Reserve an on-screen rectangle for debugging output.</summary>
+        /// <param name="client">The client caller.  This is used as a handle.</param>
+        /// <param name="text">Sample text, for determining rectangle size</param>
+        /// <param name="style">What style will be used to draw, used here for
+        /// determining rect size</param>
+        /// <returns>An area on the game screen large enough to print the text
+        /// in the style indicated</returns>
+        public static Rect GetScreenPos(Camera camera, string text, GUIStyle style)
+        {
+            var size = style.CalcSize(new GUIContent(text));
+            var pos = Vector2.zero;
+            if (camera != null)
+            {
+                var r = camera.rect; 
+                pos.x += r.x * Screen.width;
+                pos.y += (1 - r.y - r.height) * Screen.height;
+            }
+            return new Rect(pos, size);
+        }
+#endif
+
+        
+#if UNITY_EDITOR && CINEMACHINE_UIELEMENTS
+        const string k_DebugUIName = "CinemachineDebugUI";
+        static Dictionary<Camera, VisualElement> s_CameraViewRectContainers = new();
+        static GameObject s_UIDocumentHolder;
+        static UIDocument s_UIDocument;
+        public static VisualElement CreateRuntimeUIContainer(Camera outputCamera)
+        {
+            if (s_UIDocumentHolder == null)
+            {
+                s_UIDocumentHolder = GameObject.Find(k_DebugUIName);
+                if (s_UIDocumentHolder == null)
+                {
+                    s_UIDocumentHolder = new GameObject(k_DebugUIName)
+                    {
+                        hideFlags = HideFlags.NotEditable | HideFlags.HideInHierarchy,
```

Comment:
Missing DontSave flag - very important

Reply:
Why?

Reply:
UITK team said that it is preferable to have the UI document set in editor, because it needs to set some shaders before playing.

I tried with DontSave option, but then the result is inconsistent. Sometimes, we get UI, sometimes we don't.

Reply:
DontSave will prevent the object from being saved with the scene.  There is no reason to save it with the scene - it's a volatile thing that we should create it on an as-needed basis.  We do this elsewhere in CM.  Look in the code for `HideFlags.HideAndDontSave`

> UITK team said that it is preferable to have the UI document set in editor, because it needs to set some shaders before playing.

I don't see the relevance of that